### PR TITLE
Bump version to 0.12.0

### PR DESCRIPTION
**Pull request by robyoung**
- Update dependencies nrf51-hal and nrf52833-hal to 0.14.0

### CODE REVIEW

**Comment by robyoung** [comment]
@therealprof could I get a review on this one please?

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+(no changes)
+
+## [0.12.0] - 2021-11-05
```
Comment (therealprof) [inline CHANGELOG.md:12]:
```suggestion
## [0.12.0] - 2021-11-10
```

**Review by therealprof — APPROVED** [review]
I'm okay with this but maybe we should adjust the date? ;)

**Pull request merged**